pkg/dashboard/test/server_test.go

```diff
@@ -2273,6 +2431,531 @@ func (suite *functionEventTestSuite) sendRequestWithInvalidBody(method string, b
 	suite.mockPlatform.AssertExpectations(suite.T())
 }
 
+//
+// Api Gateway
+//
+
+type apiGatewayTestSuite struct {
+	dashboardTestSuite
+}
+
+func (suite *apiGatewayTestSuite) TestGetDetailSuccessful() {
+	returnedAPIGateway := platform.AbstractAPIGateway{}
+
+	name := "agw1"
+	namespace := "some-namespace"
+
+	returnedAPIGateway.APIGatewayConfig.Meta.Name = name
+	returnedAPIGateway.APIGatewayConfig.Meta.Namespace = namespace
+
+	returnedAPIGateway.APIGatewayConfig.Spec.Name = name
+	returnedAPIGateway.APIGatewayConfig.Spec.Host = "some-host"
+	returnedAPIGateway.APIGatewayConfig.Spec.Description = "some-desc"
+	returnedAPIGateway.APIGatewayConfig.Spec.Path = "some-path"
+	returnedAPIGateway.APIGatewayConfig.Spec.Upstreams = []platform.APIGatewayUpstreamSpec{
```

Comment:
style suggestion  - why aren't you populating those on creation in line 2443 ?
IMO it will be more readable. You should also be able to leave spec.name empty and have it autofilled